### PR TITLE
Remove old feature flag data

### PR DESCRIPTION
**Pull request by nickstenning**
When a feature flag is removed from the codebase, it will remain in the database. This could potentially cause very surprising issues in the event that a feature flag with the same name (but a different meaning) is added at some point in the future.

This commit adds a function which removes unknown feature flags from the database, and configures it to run at application startup.

### CODE REVIEW

File: h/features.py [hunk 1]
```diff
@@ -108,8 +142,8 @@ def features_status(request):
 
 
 def includeme(config):
+    # Remove old feature flags from the database on startup
+    config.action(None, remove_old_flags, order=90)
```
Comment (robertknight) [inline h/features.py]: Is there a central place that the order values for configuration actions could be defined, to make it more explicit what the groupings of actions are and in what order those groups run?

File: h/features.py [hunk 2]
```diff
@@ -66,6 +72,9 @@ def get_by_name(cls, name):
         """Fetch a flag by name."""
         return cls.query.filter(cls.name == name).first()
 
+    def __repr__(self):
+        return '<Feature %s everyone=%s>' % (self.name, self.everyone)
```
Comment (seanh) [inline h/features.py]: I'm curious why you still use `%s` string formatting? It's supposed to be [going away eventually in Python 3](https://docs.python.org/3.2/tutorial/inputoutput.html#old-string-formatting), isn't it?

Comment (nickstenning) [inline h/features.py]: Old habits die hard.

File: h/features.py (hunk 1 above)
Comment (nickstenning) [inline h/features.py]: It's not supposed to be an absolute ordering, and pyramid modules are supposed to be (mostly) independent of one another, so no. I'm not sure the value of such a thing would outweigh the indirection costs at the moment.

**Pull request merged**